### PR TITLE
Devtools fixes 

### LINKED ISSUE #53704: Virtual scroll positioning
### Is this a bug report or a feature request?

- [X] Bug Report
- [ ] Feature Request

### Please provide the steps to reproduce the issue [Bug Report only]

I wish I knew the steps. I just use the devtools, click around, use "inspect element" etc.
I also use Redux Devtools. Not frequently, but it's installed.

### Please provide the expected behavior vs the actual behavior you encountered [Bug Report only]

I expect to not have layout issues with the devtools. However often times it ends up looking like this.
When I scroll, the tree is scrolled but the visible area stays the same. I suppose you have issues with virtual scroller.
Works fine after reload.

### Please provide a screenshot if possible [Bug Report only]

image

### Please provide the exception or error you saw [Bug Report only]

_No response_

### Is this a browser-specific issue? If so, please specify the device, browser, and version. [Bug Report only]

```true
I did not test in other browsers. This issue has been observed for a while now in Chrome.
So Windows 10/11, Chrome any recent version.
```

### Description [Feature Request only]

_No response_

### Proposed solution [Feature Request only]

_No response_

### Alternatives considered [Feature Request only]

_No response_

### PR DESCRIPTION
**Pull request by JeanMeche**
See individual commits. 

fixes #53704

### CODE REVIEW

File: devtools/projects/ng-devtools/src/lib/devtools-tabs/directive-explorer/directive-forest/directive-forest.component.ts [hunk 1]
```diff
@@ -344,6 +344,10 @@ export class DirectiveForestComponent {
     }
   }
 
+  stopPropagation(event: Event): void {
+    event.stopPropagation();
+  }
```
Comment (dgp1130) [inline devtools/projects/ng-devtools/src/lib/devtools-tabs/directive-explorer/directive-forest/directive-forest.component.ts:349]: Question: Is this used anywhere? Was this referenced in the template somewhere and dropped from this commit?

File: devtools/projects/ng-devtools/src/lib/devtools-tabs/directive-explorer/directive-forest/directive-forest.component.html [hunk 2]
```diff
@@ -21,53 +21,55 @@
...
+
+        @switch (node.hydration?.status) {
+          @case ('hydrated') {
+            <mat-icon matTooltip="Hydrated" class="hydration">water_drop</mat-icon>
+          }
+          @case ('skipped') {
+            <mat-icon matTooltip="Skipped" class="hydration skipped">invert_colors_off</mat-icon>
+          }
+          @case ('mismatched') {
+            <mat-icon matTooltip="Mismatch" class="hydration mismatched">error_outline</mat-icon>
```
Comment (dgp1130) [inline devtools/projects/ng-devtools/src/lib/devtools-tabs/directive-explorer/directive-forest/directive-forest.component.html:53]: Suggestion: One of my points of feedback was that it's not super clear what was "Mismatched", especially for users not that familiar with hydration. Should we update these tool tips to actually use the word "hydration" such as "Hydration skipped" and "Hydration mismatch"? That way a developer would have something meaningful to search for if they want to understand what exactly this means.

**Review by dgp1130 — CHANGES_REQUESTED** [review]
Thanks for putting this together and apologies for the review delay. No major concerns, I think you just dropped a file in your `stopPropagation` commit.

@AleksanderBodurri, can you take a quick look as well just to make sure I'm not missing anything?

File: devtools/projects/ng-devtools/src/lib/devtools-tabs/directive-explorer/directive-forest/directive-forest.component.ts (hunk 1 above)
Comment (JeanMeche) [reply devtools/projects/ng-devtools/src/lib/devtools-tabs/directive-explorer/directive-forest/directive-forest.component.ts:349]: Yeah I'm not sure why the change was missing, It's back now.

**Review by dgp1130 — APPROVED** [review]
LGTM.

My one other minor suggestion is a brief comment either on `(dblclick)` or `stopPropogation` about the action it's trying to prevent, since it's purpose doesn't seem super obvious to me.

**Review by AleksanderBodurri — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
@JeanMeche the merge tooling failed to process this PR. Could you please rebase on top of the most recent `main` and merge fixup commits into corresponding parent commits? Merging fixup commits is automatic in the tooling, but for some reasons failed for this PR.

**Comment by JeanMeche** [comment]
We should be ✅ now

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 96972b45ec919a77e6031d3d0eb479905b4a699b.

**Pull request closed without merging**